### PR TITLE
BXC-4636 source_files add command

### PR DESCRIPTION
**Pull request by krwong**
[https://unclibrary.atlassian.net/browse/BXC-4636](https://unclibrary.atlassian.net/browse/BXC-4636)

- `SourceFilesCommand`: `add` command populates a source_files mapping with files from the filesystem
- `SourceFileMappingOptions`: `--extensions` option that allows user to provide list of file extensions to include and `--optional-prefix` option that allows user to add a prefix to chompb ids
- `SourceFilesService`: `addMapping` method, helper methods `gatherFilesystemCandidatePaths`, `findHighestId`, `parseIdValueToInt`
- add tests and test resources (2 tif, 1 jpeg)

### CODE REVIEW

**Review by bbpennel — CHANGES_REQUESTED** [review]

File: src/main/java/edu/unc/lib/boxc/migration/cdm/options/LowerCaseConverter.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package edu.unc.lib.boxc.migration.cdm.options;
+
+import picocli.CommandLine.ITypeConverter;
+
+public class LowerCaseConverter implements ITypeConverter<String> {
```
Comment (bbpennel) [inline src/main/java/edu/unc/lib/boxc/migration/cdm/options/LowerCaseConverter.java:5]: Add brief javadoc to describe what this thing does

File: src/main/java/edu/unc/lib/boxc/migration/cdm/options/SourceFileMappingOptions.java [hunk 2]
```diff
@@ -11,62 +10,17 @@
...
-                    "With the template: 00$1_op0$2_0001_e.tif"},
-            defaultValue = "$1")
-    private String filenameTemplate;
-
-    @Option(names = {"-l", "--lower-template"},
-            description = "Convert the filename produced from the --file-temp option to lowercase "
-                    + "prior to attempting to match against source files.")
-    private boolean lowercaseTemplate;
-
     @Option(names = {"-u", "--update"},
```
Comment (bbpennel) [inline src/main/java/edu/unc/lib/boxc/migration/cdm/options/SourceFileMappingOptions.java:20]: I guess we don't actually need this one for the Add options

File: src/main/java/edu/unc/lib/boxc/migration/cdm/services/SourceFileService.java [hunk 3]
```diff
@@ -427,8 +425,8 @@ private List<String> gatherFilesystemCandidatePaths(SourceFileMappingOptions opt
         Files.walkFileTree(basePath, new SimpleFileVisitor<>() {
             @Override
             public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
-                if (!Files.isDirectory(file) && options.getExtensions().contains(FilenameUtils.getExtension(file.toString()))
-                        && !origSourcePaths.contains(basePath.relativize(file).toString())) {
+                if (!Files.isDirectory(file) && !origSourcePaths.contains(FilenameUtils.getName(file.toString()))
```
Comment (bbpennel) [inline src/main/java/edu/unc/lib/boxc/migration/cdm/services/SourceFileService.java:428]: `FilenameUtils.getName(file.toString())` I'm not sure I understand this, isn't origSourcePaths populated with absolute paths, so we'd want to check for full file paths in there?

**Review by bbpennel — CHANGES_REQUESTED** [review]

File: src/main/java/edu/unc/lib/boxc/migration/cdm/services/SourceFileService.java (hunk 3 above)
Comment (krwong) [reply src/main/java/edu/unc/lib/boxc/migration/cdm/services/SourceFileService.java:428]: origSourcePaths is populated with `originalRecord.get(2)`, which is `basePath.relativize(file).toString()`. I can populate origSourcePaths with absolute paths.

**Review by bbpennel — APPROVED** [review]

**Pull request merged**